### DISCUSSION TITLE
Interested on working on support an open source NewSQL database (CockroachDB, TiDB, Presto etc)

### DISCUSSION THREAD

**Discussion by isHuangXin** [SeaQL/summer-of-code#8] [Q&A]
Hi, My name is Huang Xin, a master of computer science from Huazhong University of Science and Technology (华中科技大学). 
    I am interested in applying for the " Support an open source NewSQL database (CockroachDB, TiDB, Presto, etc) ” project, but I have a few questions that I need to ask you.
    1. Is this project still available for application?
    2. If it’s, I would like to know if there is some more detailed information about the project (such as the current Github repository, how do I start this project, the project expects to achieve, etc).  
    Additionally, I am interested in open source projects, and I am currently completing TiDB talent plan 2.0
    Wish you have a nice day. Thank you very much for your reply~

**Comment by billy1624** [comment]
Hey @isHuangXin,

1. Yes! Actually the application is not yet open
2. Please check https://github.com/SeaQL/summer-of-code/blob/main/2022/CONTRIBUTING.md for details

Questions, suggestions and ideas are welcomed! :)

**Reply by isHuangXin** [reply]
@billy1624 Happy holidays~ I've been thinking about the roadmap of this project these days. 
I still feel confused about what the project is going to do, can you tell me in detail what functions the SQL builder and database driver are going to achieve? Do you have any suggestions for me to start this project? e.g. project Github repo.

**Reply by billy1624** [reply]
> Basically we would like to support one of the open source NewSQL databases. CockroachDB is somewhat compatible with Postgres while TiDB is somewhat compatible with MySQL, so if we go for one of these two, we will be developing value-added facilities in the SeaQL ecosystem. If we go after Presto (or a similar non-compatible database), we will have to support it's syntax (SeaQuery), protocol and other lower level aspects of the technology stack. It's entirely up to you, just don't propose a non-open-source database thanks!

You can refer to this [project idea's](https://github.com/SeaQL/summer-of-code/blob/main/2022/README.md#3-support-an-open-source-newsql-database-cockroachdb-tidb-presto-etc) description.

For GitHub repos:
- Query Builder: https://github.com/SeaQL/sea-query
- ORM that build on top of query builder: https://github.com/SeaQL/sea-orm

**Comment by billy1624** [comment]
Hey @isHuangXin, GSoC contributor application now opens. Please submit your proposal early so we can review and provide feedbacks :)

**Comment by tyt2y3** [comment]
Yeah I think we talked a bit via email. Looking forward to your proposal

**Comment by isHuangXin** [comment]
Sorry, this project might be too hard for me, I'm just a beginner of database and I'm worried that I can't finish this project.

**Comment by bhaveshnarra** [comment]
Hi @billy1624 @tyt2y3 . I'm interested in picking this up for Google Summer Of Code 2023 if you are accepting proposals for it. I am fairly comfortable with cockroachDB and Presto. I have played around with Rust before and picking it up fulltime now but I am quite comfortable with multiple other languages(JAVA, Scala, Python and JS). I have about 4 years of dev experience during college (mostly python and JS)and around 2 years of professional dev experience(Java and Scala).

Incase you are accepting proposals for it I intend to get started on my proposal and get it reviewed as fast as possible

**Comment by tyt2y3** [comment]
I don't think CockroachDB will be a project for upcoming 2023.
I am interested to learn more about the use case of Presto and will consider it.

Note: Rust experience is required

**Comment by bhaveshnarra** [comment]
Given the huge range of [connectors ](https://prestodb.io/docs/current/connector.html) presto provides, a lot of analysts and Data engineers use it. Adding support for presto would mean developers would be able to write scripts and applications that interact with Big Data Stores. For example , I could write scripts that access and join from two different Data warehouses like Big Query and Hive using the connectors and write the output to files or into a new store and maintain these scripts as code and integrate it with my CI/CD pipelines.

I have just started exploring sea-query. I can comeback with a High Level Design for this project.

P.S
I understand that experience with Rust is required, as I am learning it right now I will fork the project and implement a POC for presto. We can take this forward if you find my implementation in Rust satisfactory.
